src/client.js

```diff
@@ -121,9 +137,38 @@ export default class Client {
       }
     });
 
-    this.on('no_longer_subscribed', (event) => {
+    this.on('notify:v2.system.no_longer_subscribed*', (event, data) => {
```

Comment:
Could you explain the difference between a `no_longer_subscribed` event and a `_disconnected`? My understand is a `_disconnected` is when the WS closes/disconnects (could be network, etc). The `no_longer_subscribed` is when the API stops sending messages to the WS because the client reached their 10 limit. Does that sound right? Also, does the backend not close the WS when it sends a `no_longer_subscribed`? If it does, would it call `_disconnected` and `no_longer_subscribed` (which would explain why you have the `this.deadChannels` check in `on('_disconnected')`).

Reply:
so `no_longer_subscribed` is a message sent by the notifications service saying we've hit our channel limit and this channel is not longer valid. It then closes the websocket remotely (hence the deadChannels like you said). `_disconnected` will usually try and reconnect, but it will use the same channelId so we need to create a new channel and connect using that. The difference between `_disconnected` and `no_longer_subscribed` is `_disconnected` is the result of the websocket *actually* being disconnected, where `no_longer_subscribed` is just a message that comes across the channel. The `no_longer_subscribed` message *only* happens if you hit the limit where `_disconnected` happens every time the websocket closes. I hope that makes sense.